### DISCUSSION TITLE
How to get "active" state on Option

### DISCUSSION THREAD

**Discussion by Filippinifra** [JedWatson/react-select#5798] [Ideas]
Hello, when I declare the Option component in the "components" prop I have the need to get the active state:

```
components={{
          Option: (props) => {
            const { data, isDisabled, isFocused, isSelected, children } = props;
         ...
```

is there a way to get from props the "isActive" state or something like that?

**Comment by manjushsh** [comment]
Hi @Filippinifra, If you mean by checking if option is selected, there is a flag `isSelected` which is set to `true`

See the example screenshot below:
<img width="583" alt="image" src="https://github.com/JedWatson/react-select/assets/94426452/ee0f2af8-8b4b-47a6-9f00-564a5f878db2">

**Reply by Filippinifra** [reply]
Yea sorry I explain me in a bad way. When the dropdown menu is open, how can I understand when an option is "pressed"? (i was meaning the CSS active status)

**Reply by manjushsh** [reply]
@Filippinifra, You can get weather option is focused (on hover)  with `isFocused` prop. and `isSelected ` and `isFocused` on 'press'. Or on "press", you can check in in `onChange` method.

props when you hover:
![image](https://github.com/JedWatson/react-select/assets/94426452/04d491ac-f60a-453d-8038-1607c4d1ec24)

Props when you select:
![image](https://github.com/JedWatson/react-select/assets/94426452/cc166898-525e-46b6-acd4-c5cb77fa3a01)
